src/clients/BundleDataClient.ts

```
@@ -163,6 +164,29 @@ export class BundleDataClient {
     this.bundleTimestampCache[key] = timestamps;
   }
 
+  private async loadPersistedDataFromArweave(blockRangesForChains: number[][]): Promise<LoadDataReturnValue> {
+    if (!isDefined(this.clients?.arweaveClient)) {
+      return undefined;
+    }
+    const persistedData = await this.clients.arweaveClient.getByTopic(
+      `bundles-${blockRangesForChains}`,
+      BundleDataToPersistToDALayerTypeSS
+    );
+    // If there is no data or the data is empty, return undefined because we couldn't
+    // pull info from the Arweave persistence layer.
```

The return type expects `LoadDataReturnValue` which consists of a bunch of subtypes, so returning `undefined` seems counter-intuitive for me. There are also multiple possible causes for returning `undefined` from this function. Thoughts on encapsulating these "failures" with a reason string so the caller can know why they didn't get any persistent data?

---

we might just want to error out and let caller catch

---

this way this can be used by someone to repost to arweave in case data is missing

---

The Arweave client should output logs if it can't find any data. I've updated the return type of the function to include an optional `undefined` type.

Returning an empty value is technically within the scope of the function unless we want to throw any time we can't find potentially missing data (data that hasn't yet persisted).

---

that's good actually so if the caller hasn't set their arweave config, then this step fails and we fallback to current logic. This would be ideal